x-pack/plugins/infra/server/lib/alerting/common/utils.ts

```diff
@@ -105,5 +126,18 @@ export const getViewInAppUrlInventory = (
   };
 
   const relativeViewInAppUrl = getInventoryViewInAppUrl(parseTechnicalFields(fields, true));
-  return getViewInAppUrl(basePath, relativeViewInAppUrl);
+  return getUrl(basePath, relativeViewInAppUrl);
 };
+
+export const getViewInMetricsAppUrl = (basePath: IBasePath, spaceId: string) =>
+  getUrl(basePath, `${getSpaceUrlFragmentFromSpaceId(spaceId)}/${LINK_TO_METRICS_EXPLORER}`);
+
+export const getAlertDetailsUrl = (
+  basePath: IBasePath,
+  spaceId: string,
+  alertUuid: string | null
+) =>
+  getUrl(
+    basePath,
+    `${getSpaceUrlFragmentFromSpaceId(spaceId)}/app/observability/alerts/${alertUuid}`
+  );
```

Comment:
Feels like this could be something the `observability` plugin could provide. That way the plugin owns its routing information and the consumers can't diverge over time. What do you think?

Reply:
Would this be a use case for a [Locator](https://github.com/elastic/kibana/tree/main/src/plugins/share#locators)?

Reply:
yes, quite so :+1: thanks for the reminder, I should have thought of that

Reply:
Thank you @smith and @weltenwort , was not aware of this pattern. Will investigate and take it further.

Reply:
Decided to pick this up in a new issue: https://github.com/elastic/kibana/issues/143940